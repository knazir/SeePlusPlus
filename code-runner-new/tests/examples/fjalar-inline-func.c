// From the test suite of https://github.com/codespecs/daikon
//   daikon/tests/kvasir-tests/

inline int foo(int x) {
  return 5;
}

int main() {
  foo(3);
  return 0;
}
