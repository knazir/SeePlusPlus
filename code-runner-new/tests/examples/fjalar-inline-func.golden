rm fjalar-inline-func.vgtrace
var trace = {
  "code": "// From the test suite of https://github.com/codespecs/daikon\n//   daikon/tests/kvasir-tests/\n\ninline int foo(int x) {\n  return 5;\n}\n\nint main() {\n  foo(3);\n  return 0;\n}\n", 
  "trace": [
    {
      "event": "step_line", 
      "func_name": "main", 
      "globals": {}, 
      "heap": {}, 
      "line": 9, 
      "ordered_globals": [], 
      "stack_to_render": [
        {
          "encoded_locals": {}, 
          "frame_id": "0xFFEFFFE30", 
          "func_name": "main", 
          "is_highlighted": true, 
          "is_parent": false, 
          "is_zombie": false, 
          "ordered_varnames": [], 
          "parent_frame_id_list": [], 
          "unique_hash": "main_0xFFEFFFE30"
        }
      ], 
      "stdout": ""
    }, 
    {
      "event": "call", 
      "func_name": "foo", 
      "globals": {}, 
      "heap": {}, 
      "line": 4, 
      "ordered_globals": [], 
      "stack_to_render": [
        {
          "encoded_locals": {}, 
          "frame_id": "0xFFEFFFE30", 
          "func_name": "main", 
          "is_highlighted": false, 
          "is_parent": false, 
          "is_zombie": false, 
          "ordered_varnames": [], 
          "parent_frame_id_list": [], 
          "unique_hash": "main_0xFFEFFFE30"
        }, 
        {
          "encoded_locals": {
            "x": [
              "C_DATA", 
              "0xFFEFFFE1C", 
              "int", 
              "<UNINITIALIZED>"
            ]
          }, 
          "frame_id": "0xFFEFFFE20", 
          "func_name": "foo", 
          "is_highlighted": true, 
          "is_parent": false, 
          "is_zombie": false, 
          "ordered_varnames": [
            "x"
          ], 
          "parent_frame_id_list": [], 
          "unique_hash": "foo_0xFFEFFFE20"
        }
      ], 
      "stdout": ""
    }, 
    {
      "event": "step_line", 
      "func_name": "foo", 
      "globals": {}, 
      "heap": {}, 
      "line": 5, 
      "ordered_globals": [], 
      "stack_to_render": [
        {
          "encoded_locals": {}, 
          "frame_id": "0xFFEFFFE30", 
          "func_name": "main", 
          "is_highlighted": false, 
          "is_parent": false, 
          "is_zombie": false, 
          "ordered_varnames": [], 
          "parent_frame_id_list": [], 
          "unique_hash": "main_0xFFEFFFE30"
        }, 
        {
          "encoded_locals": {
            "x": [
              "C_DATA", 
              "0xFFEFFFE1C", 
              "int", 
              3
            ]
          }, 
          "frame_id": "0xFFEFFFE20", 
          "func_name": "foo", 
          "is_highlighted": true, 
          "is_parent": false, 
          "is_zombie": false, 
          "ordered_varnames": [
            "x"
          ], 
          "parent_frame_id_list": [], 
          "unique_hash": "foo_0xFFEFFFE20"
        }
      ], 
      "stdout": ""
    }, 
    {
      "event": "return", 
      "func_name": "foo", 
      "globals": {}, 
      "heap": {}, 
      "line": 6, 
      "ordered_globals": [], 
      "stack_to_render": [
        {
          "encoded_locals": {}, 
          "frame_id": "0xFFEFFFE30", 
          "func_name": "main", 
          "is_highlighted": false, 
          "is_parent": false, 
          "is_zombie": false, 
          "ordered_varnames": [], 
          "parent_frame_id_list": [], 
          "unique_hash": "main_0xFFEFFFE30"
        }, 
        {
          "encoded_locals": {
            "x": [
              "C_DATA", 
              "0xFFEFFFE1C", 
              "int", 
              3
            ]
          }, 
          "frame_id": "0xFFEFFFE20", 
          "func_name": "foo", 
          "is_highlighted": true, 
          "is_parent": false, 
          "is_zombie": false, 
          "ordered_varnames": [
            "x"
          ], 
          "parent_frame_id_list": [], 
          "unique_hash": "foo_0xFFEFFFE20"
        }
      ], 
      "stdout": ""
    }, 
    {
      "event": "step_line", 
      "func_name": "main", 
      "globals": {}, 
      "heap": {}, 
      "line": 10, 
      "ordered_globals": [], 
      "stack_to_render": [
        {
          "encoded_locals": {}, 
          "frame_id": "0xFFEFFFE30", 
          "func_name": "main", 
          "is_highlighted": true, 
          "is_parent": false, 
          "is_zombie": false, 
          "ordered_varnames": [], 
          "parent_frame_id_list": [], 
          "unique_hash": "main_0xFFEFFFE30"
        }
      ], 
      "stdout": ""
    }, 
    {
      "event": "return", 
      "func_name": "main", 
      "globals": {}, 
      "heap": {}, 
      "line": 11, 
      "ordered_globals": [], 
      "stack_to_render": [
        {
          "encoded_locals": {}, 
          "frame_id": "0xFFEFFFE30", 
          "func_name": "main", 
          "is_highlighted": true, 
          "is_parent": false, 
          "is_zombie": false, 
          "ordered_varnames": [], 
          "parent_frame_id_list": [], 
          "unique_hash": "main_0xFFEFFFE30"
        }
      ], 
      "stdout": ""
    }
  ]
};
